modules/components/org.wso2.carbon.appfactory.core/src/main/java/org/wso2/carbon/appfactory/core/registry/AppFacRemoteRegistryAccessService.java

```diff
@@ -531,8 +545,138 @@ private AppFactoryResource convertResourceObject( AppFacResource appFacResource)
+
+        try {
+            DeploymentConfig deploymentConfig = JDBCResourceDAO.getInstance().getDeploymentConfig(applicationKey, stage);
+            kubernetesRuntimeProvisioningService.setRuntimeProperties(runtimeProperties, deploymentConfig);
+        } catch (RuntimeProvisioningException e) {
+            String message = "Unable to create runtime property : " + key + " for application "
+                    + applicationKey + " in tenant " + tenantDomain;
+            log.error(message, e);
+        } catch (AppFactoryException e){
+            String message = "Unable to getting deployement config details from the database";
+            log.error(message, e);
+        }
+
+    }
+
+    /**
+     * Get runtime propeties from kubernetes for application
+     *
+     * @param applicationKey id of the application
+     * @param stage          stage of the application
+     * @return list of resources as runtime properties
+     */
+    @Override
+    public List<org.wso2.carbon.appfactory.core.dto.Resource> getRuntimeProperties(String applicationKey,
+            String stage, String version) {
+
+        ApplicationContext applicationContext = new ApplicationContext();
+
+        applicationContext.setCurrentStage(stage);
+        applicationContext.setId(applicationKey);
+        applicationContext.setVersion(version);
+
+        TenantInfo tenantInfo = new TenantInfo();
+
+        int tenantId = PrivilegedCarbonContext.getCurrentContext().getTenantId();
+        String tenantDomain = PrivilegedCarbonContext.getCurrentContext().getTenantDomain();
+
+        tenantInfo.setTenantId(tenantId);
+        tenantInfo.setTenantDomain(tenantDomain);
+
+        applicationContext.setTenantInfo(tenantInfo);
+
+        KubernetesRuntimeProvisioningService kubernetesRuntimeProvisioningService
+                = new KubernetesRuntimeProvisioningService(applicationContext);
+
+        List<org.wso2.carbon.appfactory.core.dto.Resource> resources
+                = new ArrayList<org.wso2.carbon.appfactory.core.dto.Resource>();
+
+        try {
+
+            List<RuntimeProperty> runtimeProperties = kubernetesRuntimeProvisioningService.getRuntimeProperties();
+            for (RuntimeProperty runtimeProperty : runtimeProperties) {
+
+                for (Map.Entry<String, String> entry : runtimeProperty.getProperties().entrySet()) {
+                    org.wso2.carbon.appfactory.core.dto.Resource resource
+                            = new org.wso2.carbon.appfactory.core.dto.Resource();
+
+                    resource.setName(entry.getKey());
+                    resource.setDescription(entry.getValue());
+                    resources.add(resource);
+                }
+
+            }
+        } catch (RuntimeProvisioningException e) {
+            String message = "Unable to retreview runtime properties for application : " + applicationKey
+                    + " tenant : " + tenantDomain;
+            log.error(message, e);
```

Comment:
Why you are not throwing any exceptions here? How does the calling method/client know whenever something is wrong?